<launch>


    <!-- MASTER NODE! -->
    <node name="run_serial_msckf" pkg="ov_msckf" type="run_serial_msckf" output="screen" clear_params="true" required="true">
<!--        <node name="run_serial_msckf" pkg="ov_msckf" type="run_serial_msckf" output="screen" clear_params="true" required="true" launch-prefix="gdb -ex run &#45;&#45;args">-->

        <!-- bag topics -->
        <param name="topic_imu"      type="string" value="/mynteye/imu/data_raw" />
        <param name="topic_camera0"  type="string" value="/mynteye/left/image_raw" />
        <param name="topic_camera1"  type="string" value="/mynteye/right/image_raw" />

        <!-- bag parameters -->
        <param name="path_bag"    type="string" value="/media/symao/My Passport/data/VIO/sf_vio/AGV-西部兴围中转场/1/2019-08-16-14-56-30.bag" />
<!--        <param name="path_bag"    type="string" value="/home/patrick/datasets/eth/V2_03_difficult.bag" />-->
<!--        <param name="path_gt"     type="string" value="$(find ov_data)/euroc_mav/V2_03_difficult.csv" />-->
        <param name="bag_start"   type="double" value="0.0" />
        <param name="bag_durr"    type="int"    value="-1" />


        <!-- world/filter parameters -->
        <param name="use_fej"                type="bool"   value="true" />
        <param name="use_imuavg"             type="bool"   value="true" />
        <param name="use_rk4int"             type="bool"   value="true" />
        <param name="use_stereo"             type="bool"   value="true" />
        <param name="calib_cam_extrinsics"   type="bool"   value="true" />
        <param name="calib_cam_intrinsics"   type="bool"   value="true" />
        <param name="calib_cam_timeoffset"   type="bool"   value="true" />
        <param name="calib_camimu_dt"        type="double" value="0.0" />
        <param name="max_clones"             type="int"    value="11" />
        <param name="max_slam"               type="int"    value="50" />
        <param name="max_cameras"            type="int"    value="2" />
        <param name="dt_slam_delay"          type="double" value="5" />
        <param name="feat_representation"    type="string" value="GLOBAL_3D" />
        <param name="init_window_time"       type="double" value="0.75" />
        <param name="init_imu_thresh"        type="double" value="1.5" />
        <rosparam param="gravity">[0.0,0.0,9.81]</rosparam>

        <!-- tracker/extractor properties -->
        <param name="use_klt"          type="bool"   value="true" />
        <param name="num_pts"          type="int"    value="300" />
        <param name="fast_threshold"   type="int"    value="10" />
        <param name="grid_x"           type="int"    value="5" />
        <param name="grid_y"           type="int"    value="3" />
        <param name="min_px_dist"      type="int"    value="10" />
        <param name="knn_ratio"        type="double" value="0.70" />

        <!-- aruco tag/mapping properties -->
        <param name="use_aruco"        type="bool"   value="false" />
        <param name="num_aruco"        type="int"    value="1024" />
        <param name="downsize_aruco"   type="bool"   value="true" />

        <!-- sensor noise values / update -->
        <param name="up_msckf_sigma_px"            type="double"   value="1" />
        <param name="up_msckf_chi2_multipler"      type="double"   value="1" />
        <param name="up_slam_sigma_px"             type="double"   value="1" />
        <param name="up_slam_chi2_multipler"       type="double"   value="1" />
        <param name="up_aruco_sigma_px"            type="double"   value="1" />
        <param name="up_aruco_chi2_multipler"      type="double"   value="1" />
        <param name="gyroscope_noise_density"      type="double"   value="0.005" />
        <param name="gyroscope_random_walk"        type="double"   value="0.001" />
        <param name="accelerometer_noise_density"  type="double"   value="0.05" />
        <param name="accelerometer_random_walk"    type="double"   value="0.01" />

        <!-- camera intrinsics -->
        <param name="cam0_is_fisheye" type="bool" value="true" />
        <param name="cam1_is_fisheye" type="bool" value="true" />
        <rosparam param="cam0_k">[368.2586902628964, 368.26665481473077, 385.05475603049416, 237.2955683743683]</rosparam>
        <rosparam param="cam0_d">[-0.020934424934450805, -0.012944485372007895, 0.014212312776554733, -0.00717295360701927]</rosparam>
        <rosparam param="cam1_k">[368.53989785788866, 368.55143943268007, 382.49048478108654, 246.3347266593535]</rosparam>
        <rosparam param="cam1_d">[-0.022043198935541575, -0.007098313013248973, 0.00554375604510885, -0.0030494669857865195]</rosparam>


        <!-- camera extrinsics -->
        <rosparam param="T_C0toI">
            [
            0.000629644,  -1, 3.77747e-05, 0.00203259,
            0.999999, 0.000629684, 0.0010759, -0.0447113,
            -0.00107592, 3.70972e-05, 0.999999, 0.0194806,
            0, 0, 0, 1
            ]
        </rosparam>
        <rosparam param="T_C1toI">
            [
            0.00366278,  -0.999991, -0.00227448, 0.00219596,
            0.999851, 0.00362389, 0.0168731, 0.0763032,
            -0.0168647, -0.00233595, 0.999855, 0.019026,
            0, 0, 0, 1
            ]
        </rosparam>


    </node>


</launch>